File: src/main/java/com/kelsos/mbrc/data/Queue.java
package com.kelsos.mbrc.data;

import org.codehaus.jackson.annotate.JsonProperty;

public class Queue {
    @JsonProperty private String type;
    @JsonProperty private String query;

    public Queue(String type, String query) {
        this.type = type;
        this.query = query;
    }

    public String getType() {
        return type;
    }

    public String getQuery() {
        return query;
    }
}

File: app/src/main/kotlin/com/kelsos/mbrc/data/Queue.kt
package com.kelsos.mbrc.data

import android.support.annotation.StringDef
import com.fasterxml.jackson.annotation.JsonProperty


class Queue(@QueueType @JsonProperty val type: String, @JsonProperty val query: String) {

  @StringDef(NEXT, LAST, NOW)
  @Retention(AnnotationRetention.SOURCE)
  annotation class QueueType

  companion object {
    const val NEXT = "next"
    const val LAST = "last"
    const val NOW = "now"
  }
}
